src/ring_theory/hahn_series.lean

```diff
@@ -719,4 +733,250 @@ dif_neg hx
 
 end valuation
 
+section
+variables (Γ) (R) [linear_order Γ] [add_comm_monoid R]
+
+/-- An infinite family of Hahn series which has a formal coefficient-wise sum.
+  The requirements for this are that the union of the supports of the series is well-founded,
+  and that only finitely many series are nonzero at any given coefficient. -/
+structure summable_family (α : Type*) :=
+(to_fun : α → hahn_series Γ R)
+(is_wf_Union_support' : set.is_wf (⋃ (a : α), (to_fun a).support))
+(co_support : Γ → finset α)
+(mem_co_support' : ∀ (a : α) (g : Γ), a ∈ co_support g ↔ (to_fun a).coeff g ≠ 0)
+
+end
+
+namespace summable_family
+section add_comm_monoid
+
+variables [linear_order Γ] [add_comm_monoid R] {α : Type*}
+
+instance : has_coe_to_fun (summable_family Γ R α) :=
+⟨λ _, (α → hahn_series Γ R), to_fun⟩
+
+lemma is_wf_Union_support (s : summable_family Γ R α) : set.is_wf (⋃ (a : α), (s a).support) :=
+s.is_wf_Union_support'
+
+@[simp]
+lemma mem_co_support {s : summable_family Γ R α} {a : α} {g : Γ} :
+  a ∈ s.co_support g ↔ (s a).coeff g ≠ 0 := mem_co_support' _ _ _
+
+@[ext]
+lemma ext {s t : summable_family Γ R α} (h : ∀ (a : α), s a = t a) : s = t :=
+begin
+  have hs := s.mem_co_support',
+  have ht := t.mem_co_support',
+  cases s,
+  cases t,
+  simp only,
+  split,
+  { ext,
+    exact congr (congr rfl (h _)) rfl },
+  { ext g a,
+    refine (hs a g).trans (iff.trans _ (ht a g).symm),
+    simp only [ne.def, not_iff_not],
+    split; intro h',
+    { exact eq.trans (congr (congr rfl (h _).symm) rfl) h' },
+    { exact eq.trans (congr (congr rfl (h _)) rfl) h' } }
+end
+
+instance : has_add (summable_family Γ R α) :=
+⟨λ x y, { to_fun := x + y,
+    is_wf_Union_support' := (x.is_wf_Union_support.union y.is_wf_Union_support).mono (begin
+      rw ← set.Union_union_distrib,
+      exact set.Union_subset_Union (λ a, support_add_subset)
+    end),
+    co_support := λ g, ((x.co_support g) ∪ (y.co_support g)).filter
+      (λ a, (x a).coeff g + (y a).coeff g ≠ 0),
+    mem_co_support' := λ a g, begin
+      simp only [mem_union, mem_filter, mem_co_support, and_iff_right_iff_imp,
+        pi.add_apply, ne.def, add_coeff'],
+      contrapose!,
+      rintro ⟨hx, hy⟩,
+      simp [hx, hy],
+    end }⟩
+
+instance : has_zero (summable_family Γ R α) :=
+⟨⟨0, by simp, λ _, ∅, by simp⟩⟩
+
+instance : inhabited (summable_family Γ R α) := ⟨0⟩
+
+@[simp]
+lemma add_apply {s t : summable_family Γ R α} {a : α} : (s + t) a = s a + t a := rfl
```

Comment:
```suggestion
@[simp]
lemma coe_add (s t : summable_family Γ R α) : ⇑(s + t) = s + t := rfl
lemma add_apply (s t : summable_family Γ R α) (a : α) : (s + t) a = s a + t a := rfl
```
and similarly for the others.

Reply:
I couldn't find a suitable form for `smul_apply`, but I did that for the others.